### PR TITLE
[MOB-11886] Add `Instabug.init` API

### PR DESCRIPTION
**Pull request by TheBuggedYRN**
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
### Development
- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
### Code review 
- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## Unreleased
+
+* Deprecates `Instabug.start` in favour of the new `Instabug.init` API.
```
Comment (DavidMina96) [inline CHANGELOG.md:3]: suggestion: I think we need to elaborate more on the usage of the new API or add a docs reference after updating it.

File: lib/src/modules/instabug.dart [hunk 2]
```diff
@@ -153,12 +153,25 @@ class Instabug {
   /// it on your dashboard.
   /// The [invocationEvents] are the events that invoke
   /// the SDK's UI.
+  static Future<void> init({
```
Comment (DavidMina96) [inline lib/src/modules/instabug.dart:156]: question: Should we add an optional parameter for `debugLogsLevel` and do the associated changes in this PR as well?

Comment (TheBuggedYRN) [reply lib/src/modules/instabug.dart:156]: This will be handled in a separate PR

File: CHANGELOG.md (hunk 1 above)
Comment (TheBuggedYRN) [reply CHANGELOG.md:3]: We can do this in before the release when all items related to `Instabug.init` are added to the Changelog, then we will work on grouping and rephrasing them together in one solid point.

**Review by DavidMina96 — APPROVED** [review]

**Pull request merged**